packages/test-apps/simple-react/test/navigation-details.spec.ts

```diff
@@ -38,4 +38,18 @@ describe('navigation details', () => {
     expect(testCalls[1].isNavigating).toBe(true)
     expect(testCalls[1].navigationDestination).toBe('/navigation-test#section')
   })
```

Comment:
The change may be more involved, but should query params changing hit the server, whereas anchor tag navigation would not? This would be more challenging to write, and it won't matter for our needs right now

Reply:
It actually does do that already! The route match needs to ignore the query params, but we include them in the cache key for the JSON data we fetch from the server to render a page.